src/completion_item_map_impl.ts

```diff
@@ -0,0 +1,42 @@
+import { CompletionItemMap } from "./completion_item_map";
+import * as vscode from "vscode";
+
+/**
+ * Map from the first char of CompletionItem.label to vscode.CompletionItem[]
```

Comment:
Nit: This comment looks slightly inaccurate. It looks like the logic to decide that the key for a completion item should be the first character isn't actually in this file and is passed in as a parameter.